File: ruby/testcode/cmdi_036.rb
require_relative 'shared'

# vuln-code-snippet start ruby_cmdi_process_spawn_array
def ping_host_safe(req)
  host = req.param('host').gsub(/[^a-z0-9.\-]/i, '')
  pid = Process.spawn('ping', '-c', '1', host)  # vuln-code-snippet safe-line ruby_cmdi_process_spawn_array
  Process.wait(pid)
  BenchmarkResponse.json({ pid: pid })
end
# vuln-code-snippet end ruby_cmdi_process_spawn_array

File: php/testcode/cmdi_036.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_cmdi_filter_validate_ip
function cmdi036(BenchmarkRequest $req): BenchmarkResponse {
    $ip = $req->param('ip');
    if (!filter_var($ip, FILTER_VALIDATE_IP)) {
        return BenchmarkResponse::badRequest('invalid ip');
    }
    $output = [];
    exec("ping -c 3 " . $ip, $output); // vuln-code-snippet safe-line php_cmdi_filter_validate_ip
    return BenchmarkResponse::ok(implode("\n", $output));
}
// vuln-code-snippet end php_cmdi_filter_validate_ip
